tiledb/sm/misc/parallel_functions.h

```diff
@@ -232,7 +273,13 @@ Status parallel_for(
   // Wait for all instances of `execute_subrange` to complete.
   tp->wait_all(tasks);
 
-  return return_st;
+  if (failed_exception.has_value()) {
+    std::rethrow_exception(failed_exception.value());
+  }
```

Comment:
Is this new behavior?  Meaning, the current implementation doesn't rethrow.  I assume this is part of our migration to using exceptions rather than return values.

Reply:
Yes, this is the one new behavior I introduced. `StatusException` would be passed transparently through a `Status` return, but `std::runtime_error` etc. would not be.

Per @dhoke4tdb's report in a comment below, there's no regression with this when he's throwing inside a subrange function. It was the only thing I was particularly worried about.